srml/balances/src/lib.rs

```diff
@@ -398,17 +370,21 @@ where
 	}
 
 	fn can_reserve(who: &T::AccountId, value: Self::Balance) -> bool {
-		if T::EnsureAccountLiquid::ensure_account_liquid(who).is_ok() {
+		if T::EnsureAccountLiquid::ensure_account_can_transfer(who).is_ok() {
 			Self::free_balance(who) >= value
 		} else {
 			false
 		}
 	}
 
-	fn total_issuance() -> Self:: Balance {
+	fn total_issuance() -> Self::Balance {
 		Self::total_issuance()
```

Comment:
isn't this an infinite recursive call? or is there any mechanism to ensure this calls `total_issuance` generated by `decl_storage`?

Reply:
That's not new, but it does look dodgy...